app/app_client.go

```diff
@@ -469,28 +544,17 @@ func (c *AppClient) GetRobotPart(ctx context.Context, id string) (*RobotPart, st
 	return robotPartFromProto(resp.Part), resp.ConfigJson, nil
 }
 
-// GetRobotPartLogsOptions contains optional parameters for GetRobotPartLogs.
-type GetRobotPartLogsOptions struct {
-	Filter    *string
-	PageToken *string
-	Levels    []string
-	Start     *timestamppb.Timestamp
-	End       *timestamppb.Timestamp
-	Limit     *int64
-	Source    *string
-}
-
-// GetRobotPartLogs gets the logs associated with a robot part and the next page token,
-// defaulting to the most recent page if pageToken is empty. Logs of all levels are returned when levels is empty.
-func (c *AppClient) GetRobotPartLogs(ctx context.Context, id string, opts GetRobotPartLogsOptions) ([]*LogEntry, string, error) {
+// GetRobotPartLogs gets the logs associated with a robot part and the next page token.
+func (c *AppClient) GetRobotPartLogs(ctx context.Context, id string, opts *GetRobotPartLogsOptions) ([]*LogEntry, string, error) {
+	limit := int64(*opts.Limit)
```

Comment:
(q) is this safe? this looks to me like we could be attempting to dereference a nil value which is unsafe in most languages, but it wouldn't shock me if go has some optimization/fallback on a default value here... @benjirewis do you know?

Reply:
Based on [this](https://medium.com/@ezrantn/understanding-and-preventing-nil-pointer-dereference-panics-in-go-1b0d341fa6b0) I suspect it is not safe and we should be refactoring.

Reply:
Yeah not safe; need to check `if opts.Limit != nil` to avoid npd.